### DISCUSSION TITLE
History message

### DISCUSSION THREAD

**Discussion by Abbadon999** [HRNPH/AIwaifu#22] [General]
I don't understand, does this waifu have a message history or is every new message like a new message to it? It's just that every time I ask her a question that refers to information from previous posts, she comes up with something of her own, but she can't answer properly...

**Comment by HRNPH** [comment]
well this is due to our badly implemented caching code
First of all, you need to understand that the WHOLE input of the model was

```
long ass description of who she is (Usually 5-6 lines)
long ass description of who she is (Usually 5-6 lines)
long ass description of who she is (Usually 5-6 lines)
long ass description of who she is (Usually 5-6 lines)
long ass description of who she is (Usually 5-6 lines)
long ass description of who she is (Usually 5-6 lines)
[''Name"]:

# now our conversation start here

["YOU"] : talking and asking stuff....
["Name"] : Answer

["YOU"] : talking and asking stuff....
["Name"] : Answer

["YOU"] : talking and asking stuff....
["Name"] : Answer

["YOU"] : talking and asking stuff....
["Name"] : Answer
```
this will then be encoded into embedding which will be as long as the conversation goes on
the longer we talk the longer it'll take to inference and it'll take so much ram
Just to be safe, in development phase I capped the conversation to only 2-3 last question and will then be deleted
which Is the reason why when you ask question she answer it out of nowhere

Basically this file https://github.com/HRNPH/AIwaifu/blob/master/conversation.py
![image](https://user-images.githubusercontent.com/51855316/231877136-411dab8c-e93b-4ac6-9b5a-5dace9db3dda.png)

**Comment by HRNPH** [comment]
AKA just change that number

it'll be more configurable via the config file, After I set up all the components for TTS this will be next
